charts/gateway-helm/templates/envoy-gateway-hpa.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if .Values.hpa.enabled }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: envoy-gateway
+  namespace: '{{ $.Release.Namespace }}'
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: envoy-gateway
+  {{- if .Values.hpa.minReplicas }}
+  minReplicas: {{ .Values.hpa.minReplicas }}
+  {{- end }}
+  maxReplicas: {{ .Values.hpa.maxReplicas }}
```

Comment:
```suggestion
  {{- if .Values.hpa.maxReplicas }}
  maxReplicas: {{ .Values.hpa.maxReplicas }}
  {{- end }}
```

missing?

Reply:
I can't seem to find the resource I read at the time but I did this because `maxReplicas` is a required value.

Just double checked:
```yaml
apiVersion: autoscaling/v2
kind: HorizontalPodAutoscaler
metadata:
  name: envoy-gateway
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: envoy-gateway
  # maxReplicas: 1
```


```
$ kubectl apply -f hpa.yaml 
The HorizontalPodAutoscaler "envoy-gateway" is invalid: 
* spec.maxReplicas: Invalid value: 0: must be greater than 0
```

I think it's possible to set a default value in gotmpl or document if you want.

Reply:
Or should I add the if as suggested and then I think helm should surface the error?

Reply:
can we use `required` then?
https://helm.sh/docs/howto/charts_tips_and_tricks/

Reply:
how do we implement other template values for required fields? (do we have such?)

Reply:
I don't see any other examples in this chart but yes that works. 
```shell
$ helm template .
Error: execution error at (gateway-helm/templates/envoy-gateway-hpa.yaml:15:18): .Values.hps.maxRetries entry is required

Use --debug flag to render out invalid YAML

```
And the `maxRetries` value is ignored if `hpa.enabled=false` as the template is not rendered.